js/huobi.js

```diff
@@ -4617,6 +4618,128 @@ module.exports = class huobi extends Exchange {
         return this.filterByArray (result, 'symbol', symbols);
     }
 
+    async fetchBorrowInterest (code = undefined, symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const defaultMargin = this.safeString (params, 'marginType', 'cross'); // cross or isolated
+        const marginType = this.safeString2 (this.options, 'defaultMarginType', 'marginType', defaultMargin);
+        const request = {};
+        if (since !== undefined) {
+            request['start-date'] = this.yyyymmdd (since);
+        }
+        if (limit !== undefined) {
+            request['size'] = limit;
+        }
+        let market = undefined;
+        let method = undefined;
+        if (marginType === 'isolated') {
+            method = 'privateGetMarginLoanOrders';
+            if (symbol !== undefined) {
+                market = this.market (symbol);
+                request['symbol'] = market['id'];
```

Comment:
If this is way we do this method, then symbol must be defined, so I thought it was just easier to infer that it's isolated if a symbol is defined, rather than get a margin type, and then throw an error if symbol isn't defined

![](https://i.imgur.com/fir8GyM.png)

Reply:
@samgermain although the docs state that, it's not true you can do the request without it

Reply:
@samgermain Example:

```
 node cli.js huobi fetchBorrowInterest --verbose undefined undefined undefined undefined '{"marginType":"isolated"}'
Debugger attached.
2022-04-07T13:09:23.496Z
Node.js: v17.6.0
CCXT v1.78.36
huobi.fetchBorrowInterest (, , , , [object Object])
fetch Request:
 huobi GET https://api.huobi.pro/v1/margin/loan-orders?AccessKeyId=ez2xc4vb6n-da8c4c7d-76cbde5b-7cc77&SignatureMethod=HmacSHA256&SignatureVersion=2&Timestamp=2022-04-07T13%3A09%3A29&marginType=isolated&Signature=x%2Ft%2BalXGA0IEa8CqOUREBN5YxTs4ZGsLR9o9VZ%2FhZ%2FQ%3D 
RequestHeaders:
 {
  'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36',
  'Content-Type': 'application/x-www-form-urlencoded'
} 
RequestBody:
 undefined 

handleRestResponse:
 huobi GET https://api.huobi.pro/v1/margin/loan-orders?AccessKeyId=ez2xc4vb6n-da8c4c7d-76cbde5b-7cc77&SignatureMethod=HmacSHA256&SignatureVersion=2&Timestamp=2022-04-07T13%3A09%3A29&marginType=isolated&Signature=x%2Ft%2BalXGA0IEa8CqOUREBN5YxTs4ZGsLR9o9VZ%2FhZ%2FQ%3D 200 OK 

ResponseBody:
 {"status":"ok","data":[{"interest-rate":"0.000040830000000000","loan-amount":"100.790000000000000000","user-id":35930539,"accrued-at":1649320794148,"created-at":1649320794148,"interest-amount":"0.004115260000000000","updated-at":1649322538837,"symbol":"ltcusdt","currency":"usdt","paid-coin":"0.004115260000000000","loan-balance":"0.000000000000000000","interest-balance":"0.000000000000000000","account-id":48916071,"deduct-amount":"0","deduct-currency":"","deduct-rate":"1","paid-point":"0.000000000000000000","day-interest-rate":"0.000980000000000000","hour-interest-rate":"0.000040830000000000","id":20242721,"state":"cleared"}],"code":200} 

2022-04-07T13:09:32.603Z iteration 0 passed in 6702 ms

 account | currency |   interest | interestRate | amountBorrowed |     timestamp |                 datetime
-----------------------------------------------------------------------------------------------------------
LTC/USDT |     USDT | 0.00411526 |   0.00004083 |         100.79 | 1649320794148 | 2022-04-07T08:39:54.148Z
```